app/src/main/java/com/example/benchproject/app/popular/movies/view/PopularMoviesRecyclerAdapter.kt

```diff
@@ -6,9 +6,10 @@ import android.view.ViewGroup
 import androidx.recyclerview.widget.RecyclerView
 import com.example.benchproject.app.common.binding.imageView.loadMovieImage
 import com.example.benchproject.app.popular.movies.entity.Movie
+import com.example.benchproject.app.popular.movies.viewModel.PopularMoviesViewModel
 import com.example.benchproject.databinding.PopularMovieRecyclerItemLayoutBinding
 
-class PopularMoviesRecyclerAdapter : RecyclerView.Adapter<PopularMoviesRecyclerAdapter.PopularMoviesHolder>() {
+class PopularMoviesRecyclerAdapter(val moviesViewModel: PopularMoviesViewModel) : RecyclerView.Adapter<PopularMoviesRecyclerAdapter.PopularMoviesHolder>() {
```

Comment:
This should be private and I changed it. I used this `moviesViewModel` only in this adapter and I added it by the constructor to avoid any extra arguments in the function `uploadMovieData` where I upload a list